pilot/pkg/bootstrap/istio_ca.go

```diff
@@ -434,3 +446,17 @@ func (s *Server) createIstioRA(client kubelib.Client,
 	return ra.NewIstioRA(raOpts)
 
 }
+
+func kubeDefaultSignerSupported(client kubelib.Client) (bool, string) {
+	if client == nil {
+		return false, ""
+	}
+	version, err := k8sversion.GetKubernetesVersion(client.RESTConfig())
+	if err != nil {
+		return false, ""
```

Comment:
temporary 1s downtime of api-server --> implicitly switching to self-signed CA will cause both outage and is probably a massive security issue.

Reply:
I would much prefer to do the check in istioctl..

Reply:
Is there any Istio config validation today that istioctl does based on the Kubernetes version? Went through the code and didn't seem to find any. Most of the validation checks appear to be prior to install time when istioctl finally determines the K8s version on the cluster.  Didn't want to introduce install time validation checks in istioctl unless it was absolutely required.

Reply:
https://github.com/istio/istio/blob/96f0b401cdc7794817187334bcbee44ff1193d48/operator/pkg/manifest/shared.go#L382

Reply:
https://github.com/istio/istio/pull/30368 - the check is implemented as part of istioctl